File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.cpp
// Last updated: 8/4/2026, 7:21:42 PM
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        int min_price = prices[0] , profit = 0;
        for(int i = 0 ; i < n ; i++){
            int cost = prices[i] - min_price;
            profit = max(profit , cost);
            min_price = min(min_price , prices[i]);
        }
        return profit;
    }
};

File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.java
// Last updated: 8/4/2026, 7:21:49 PM
class Solution {
    public int maxProfit(int[] prices) {
        int n = prices.length;
        int buy = prices[0];
        int profit = 0;

        for(int i = 1 ; i < n ; i++){
            if(prices[i] > buy){
                profit = Math.max(profit , prices[i] - buy);
            }
            buy = Math.min(buy , prices[i]);
        }
        return profit;
    }
}